onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -731,7 +731,7 @@ class PlannerImpl {
           // we do not try sharing-optimization for non-tensors
           AllocPlan(current).alloc_kind = AllocKind::kAllocate;
           AllocPlan(current).program_counter.AddStart(program_counter);
-        } else if (FindReusableInput(*pnode, static_cast<int>(output_arg_def_index), &reused)) {
+        } else if (!context_.IsParallelExecutionEnabled() && FindReusableInput(*pnode, static_cast<int>(output_arg_def_index), &reused)) {
           // Reuse one of this node's input buffers as the output buffer (for in-place update)
```

Comment:
Can we instead change FindReusableInput to look at the edges for the input it's considering re-using and require there to be only one if parallel execution is enabled?

Reply:
Do you know why it works in sequential mode?

Reply:
I read the code again. It is good in sequential mode because the planner tracks use count. And when ever a reuse happens, it will increase the use count. But all such things based on an assumption that the graph will be executed in topological sort order. For example, the parent loop of the line 733 iterates all the nodes in topological sort order. It doesn't make sense for parallel execution.  Even through we can do some improvements here, I feel it's not worth. We need to change the logic to track edge count instead of tracking usage count till current program_counter. I think we should create a new planner class instead.